sdks/java/core/src/main/java/org/apache/beam/sdk/util/Credentials.java

```diff
@@ -132,15 +135,32 @@ public static Credential getCredential(GcpOptions options)
    * Loads OAuth2 credential from a local file.
    */
   private static Credential getCredentialFromFile(
-      String keyFile, String accountId, Collection<String> scopes)
+          @Nullable String keyFile, @Nullable String accountName, Collection<String> scopes)
       throws IOException, GeneralSecurityException {
-    GoogleCredential credential = new GoogleCredential.Builder()
+
+    if (keyFile == null) {
+      throw new IOException("If accountName is given, also supply a keyFile");
+    }
```

Comment:
that doesn't cover all scenario's. But I think I a pushed something that covers all remarks:

In: 

- getCredential(GcpOptions options) -> keyFile or accountName is give -> check at least that the keyFile is given (required for JSON and P12)
- getCredentialFromFile -> check the JSON case (no accountName allowed) or 12 case (accountName required)

Because keyFile is checked of in getCredential @Nullable could be removed from keyFile in getCredentialFromFile.